lib/rich_text.rb

```diff
@@ -87,17 +88,65 @@ def sanitize(text)
     end
 
     def linkify(text, mode = :urls)
-      link_attr = 'rel="nofollow noopener noreferrer" dir="auto"'
-      Rinku.auto_link(ERB::Util.html_escape(text), mode, link_attr) do |url|
-        url = shorten_host(url, Settings.linkify_hosts, Settings.linkify_hosts_replacement)
-        shorten_host(url, Settings.linkify_wiki_hosts, Settings.linkify_wiki_hosts_replacement) do |path|
-          path.sub(Regexp.new(Settings.linkify_wiki_optional_path_prefix || ""), "")
-        end
-      end.html_safe
+      ERB::Util.html_escape(text)
+               .then { |html| expand_link_shorthands(html) }
+               .then { |html| expand_host_shorthands(html) }
+               .then { |html| auto_link(html, mode) }
+               .html_safe
     end
 
     private
 
+    def gsub_pairs_for_linkify_detection
+      Array
+        .wrap(Settings.linkify&.detection_rules)
+        .select { |rule| rule.path_template && rule.patterns.is_a?(Array) }
+        .flat_map do |rule|
+          expanded_path = "#{rule.host || "#{Settings.server_protocol}://#{Settings.server_url}"}/#{rule.path_template}"
+          rule.patterns
+              .select { |pattern| pattern.is_a?(String) }
+              .map do |pattern|
+            [
+              Regexp.new("(?<before>^|#{URL_UNSAFE_CHARS})#{pattern}", Regexp::IGNORECASE),
+              "\\k<before>#{expanded_path}"
```

Comment:
I think you could use a lookbehind assertion here and then you wouldn't need to capture the prefix and add it back, like this:

```suggestion
              Regexp.new("(?<=^|#{URL_UNSAFE_CHARS})#{pattern}", Regexp::IGNORECASE),
              expanded_path
```